zone/tradeskills.cpp

```diff
@@ -468,23 +468,42 @@ void Object::HandleCombine(Client* user, const NewCombine_Struct* in_combine, Ob
 		}
 	}
 
+	//pull the list of components so we can compare during lore check
+	std::string query = fmt::format("SELECT tre.item_id "
+	                                 "FROM tradeskill_recipe_entries AS tre "
+	                                 "WHERE tre.componentcount > 0 AND tre.recipe_id = {}",
+	                                 spec.recipe_id);
+	auto results = content_db.QueryDatabase(query);
+
+	if (!results.Success()) {
+		auto outapp = new EQApplicationPacket(OP_TradeSkillCombine, 0);
+		user->QueuePacket(outapp);
+		safe_delete(outapp);
+		return;
+	}
+
+	if(results.RowCount() < 1) {
+		LogTradeskills("Error in HandleCombine: no components returned");
+		auto outapp = new EQApplicationPacket(OP_TradeSkillCombine, 0);
+		user->QueuePacket(outapp);
+		safe_delete(outapp);
+		return;
+	}
```

Comment:
Can these be consolidated since they'd both be an error?

Reply:
Yeah, I just merged all these checks, it's just verbose for no real gain.